In [12]:
#Importing relevant libraries
import pandas as pd
import matplotlib as plt

In [13]:
#Importing dataframe
animes = pd.read_csv('anime.csv')
#Filtering out Unknown scores
animes = animes[animes['Score'] != 'Unknown']
#Sorting by score, higher to lower
animes = animes.sort_values(by = 'Score', ascending = False)
#First five elements
animes.head()

MAL_ID                                  Name Score  \
3971     5114      Fullmetal Alchemist: Brotherhood  9.19   
15926   40028  Shingeki no Kyojin: The Final Season  9.17   
5683     9253                           Steins;Gate  9.11   
14963   38524    Shingeki no Kyojin Season 3 Part 2   9.1   
9913    28977                              Gintama°   9.1   

                                                  Genres  \
3971   Action, Military, Adventure, Comedy, Drama, Ma...   
15926  Action, Military, Mystery, Super Power, Drama,...   
5683                                    Thriller, Sci-Fi   
14963  Action, Drama, Fantasy, Military, Mystery, Sho...   
9913   Action, Comedy, Historical, Parody, Samurai, S...   

                          English name               Japanese name Type  \
3971   Fullmetal Alchemist:Brotherhood  鋼の錬金術師 FULLMETAL ALCHEMIST   TV   
15926     Attack on Titan Final Season      進撃の巨人 The Final Season   TV   
5683                       Steins;Gate                 STEINS;GATE   TV   
14963  Attack on Titan Season 3 Part 2        進撃の巨人 Season3 Part.2   TV   
9913                  Gintama Season 4                         銀魂°   TV   

      Episodes                        Aired    Premiered  ...  Score-10  \
3971        64   Apr 5, 2009 to Jul 4, 2010  Spring 2009  ...  714811.0   
15926       16             Dec 7, 2020 to ?  Winter 2021  ...  173154.0   
5683        24  Apr 6, 2011 to Sep 14, 2011  Spring 2011  ...  468504.0   
14963       10  Apr 29, 2019 to Jul 1, 2019  Spring 2019  ...  327290.0   
9913        51  Apr 8, 2015 to Mar 30, 2016  Spring 2015  ...   78918.0   

        Score-9   Score-8  Score-7  Score-6  Score-5 Score-4  Score-3  \
3971   401507.0  199160.0  70045.0  20210.0   9308.0  3222.0   1536.0   
15926   63756.0   26016.0   8793.0   2674.0   1336.0   588.0    382.0   
5683   275960.0  140914.0  57740.0  21375.0  11126.0  5061.0   2292.0   
14963  239451.0  110481.0  33662.0   8365.0   2974.0  1108.0    550.0   
9913    37902.0   21360.0  10215.0   3898.0   2311.0   952.0    648.0   

       Score-2  Score-1  
3971    2162.0  16806.0  
15926    514.0  11061.0  
5683    1678.0   5255.0  
14963    385.0   4169.0  
9913    1100.0   4508.0  

[5 rows x 35 columns]

In [14]:
# Filtering out all problematic data, like Unknown values in needed to be numeric columns
animes['Score-1'] = animes['Score-1'].astype('object')
a1_list = animes['Score-1'] != 'Unknown'
animes = animes[a1_list]
a2_list = animes['Score-2'] != 'Unknown'
animes = animes[a2_list]
a3_list = animes['Score-3'] != 'Unknown'
animes = animes[a3_list]
a4_list = animes['Score-4'] != 'Unknown'
animes = animes[a4_list]
a5_list = animes['Score-5'] != 'Unknown'
animes = animes[a5_list]
a6_list = animes['Score-6'] != 'Unknown'
animes = animes[a6_list]
a7_list = animes['Score-7'] != 'Unknown'
animes = animes[a7_list]
a8_list = animes['Score-8'] != 'Unknown'
animes = animes[a8_list]
a9_list = animes['Score-9'] != 'Unknown'
animes = animes[a9_list]
a10_list = animes['Score-10'] != 'Unknown'
animes = animes[a10_list]
animes.head()

MAL_ID                                  Name Score  \
3971     5114      Fullmetal Alchemist: Brotherhood  9.19   
15926   40028  Shingeki no Kyojin: The Final Season  9.17   
5683     9253                           Steins;Gate  9.11   
14963   38524    Shingeki no Kyojin Season 3 Part 2   9.1   
9913    28977                              Gintama°   9.1   

                                                  Genres  \
3971   Action, Military, Adventure, Comedy, Drama, Ma...   
15926  Action, Military, Mystery, Super Power, Drama,...   
5683                                    Thriller, Sci-Fi   
14963  Action, Drama, Fantasy, Military, Mystery, Sho...   
9913   Action, Comedy, Historical, Parody, Samurai, S...   

                          English name               Japanese name Type  \
3971   Fullmetal Alchemist:Brotherhood  鋼の錬金術師 FULLMETAL ALCHEMIST   TV   
15926     Attack on Titan Final Season      進撃の巨人 The Final Season   TV   
5683                       Steins;Gate                 STEINS;GATE   TV   
14963  Attack on Titan Season 3 Part 2        進撃の巨人 Season3 Part.2   TV   
9913                  Gintama Season 4                         銀魂°   TV   

      Episodes                        Aired    Premiered  ...  Score-10  \
3971        64   Apr 5, 2009 to Jul 4, 2010  Spring 2009  ...  714811.0   
15926       16             Dec 7, 2020 to ?  Winter 2021  ...  173154.0   
5683        24  Apr 6, 2011 to Sep 14, 2011  Spring 2011  ...  468504.0   
14963       10  Apr 29, 2019 to Jul 1, 2019  Spring 2019  ...  327290.0   
9913        51  Apr 8, 2015 to Mar 30, 2016  Spring 2015  ...   78918.0   

        Score-9   Score-8  Score-7  Score-6  Score-5 Score-4  Score-3  \
3971   401507.0  199160.0  70045.0  20210.0   9308.0  3222.0   1536.0   
15926   63756.0   26016.0   8793.0   2674.0   1336.0   588.0    382.0   
5683   275960.0  140914.0  57740.0  21375.0  11126.0  5061.0   2292.0   
14963  239451.0  110481.0  33662.0   8365.0   2974.0  1108.0    550.0   
9913    37902.0   21360.0  10215.0   3898.0   2311.0   952.0    648.0   

       Score-2  Score-1  
3971    2162.0  16806.0  
15926    514.0  11061.0  
5683    1678.0   5255.0  
14963    385.0   4169.0  
9913    1100.0   4508.0  

[5 rows x 35 columns]

In [4]:
#Info on columns, names, non-null count and data type
animes.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 12116 entries, 3971 to 2934
Data columns (total 35 columns):
 #   Column         Non-Null Count  Dtype 
---  ------         --------------  ----- 
 0   MAL_ID         12116 non-null  int64 
 1   Name           12116 non-null  object
 2   Score          12116 non-null  object
 3   Genres         12116 non-null  object
 4   English name   12116 non-null  object
 5   Japanese name  12116 non-null  object
 6   Type           12116 non-null  object
 7   Episodes       12116 non-null  object
 8   Aired          12116 non-null  object
 9   Premiered      12116 non-null  object
 10  Producers      12116 non-null  object
 11  Licensors      12116 non-null  object
 12  Studios        12116 non-null  object
 13  Source         12116 non-null  object
 14  Duration       12116 non-null  object
 15  Rating         12116 non-null  object
 16  Ranked         12116 non-null  object
 17  Popularity     12116 non-null  int64 
 18  Members        12116 non

# Objectives of this project:
### -Find a good underrated anime to watch
### -Find the most overrated anime
### -Figure out if the most hated anime(not by score)
### -Figure out the most loved anime(not by score)

In [15]:
#Converting data types to make it more manageable
animes['Score'] = animes['Score'].astype('float64')
animes['Score-10'] = animes['Score-10'].astype('float64')
animes['Score-9'] = animes['Score-9'].astype('float64')
animes['Score-8'] = animes['Score-8'].astype('float64')
animes['Score-7'] = animes['Score-7'].astype('float64')
animes['Score-6'] = animes['Score-6'].astype('float64')
animes['Score-5'] = animes['Score-5'].astype('float64')
animes['Score-4'] = animes['Score-4'].astype('float64')
animes['Score-3'] = animes['Score-3'].astype('float64')
animes['Score-2'] = animes['Score-2'].astype('float64')
animes['Score-1'] = animes['Score-1'].astype('float64')
animes.dtypes

MAL_ID             int64
Name              object
Score            float64
Genres            object
English name      object
Japanese name     object
Type              object
Episodes          object
Aired             object
Premiered         object
Producers         object
Licensors         object
Studios           object
Source            object
Duration          object
Rating            object
Ranked            object
Popularity         int64
Members            int64
Favorites          int64
Watching           int64
Completed          int64
On-Hold            int64
Dropped            int64
Plan to Watch      int64
Score-10         float64
Score-9          float64
Score-8          float64
Score-7          float64
Score-6          float64
Score-5          float64
Score-4          float64
Score-3          float64
Score-2          float64
Score-1          float64
dtype: object

In [30]:
#Defining 4 dataframes to work each objective in each one of them
animeso1 = animes
animeso2 = animes
animeso3 = animes
animeso4 = animes

## Fulfilling objective 1
In order to fulfill the objective i am thinking of finding an anime with a very small amount of total votes in comparison with the other animes that still has a great score

In [7]:
#For the first step, ill create a new table that contains the total amount of votes
animeso1['Total Votes'] = animeso1['Score-10'] + animeso1['Score-9'] +animeso1['Score-8'] + animeso1['Score-7'] + animeso1['Score-6'] + animeso1['Score-5'] + animeso1['Score-4'] + animeso1['Score-3'] + animeso1['Score-2'] + animeso1['Score-1']
animeso1.iloc[:,[1,2,-1,-2,-3,-4,-5,-6,-7,-8,-9,-10,-11]]

Name  Score  Total Votes  Score-1  \
3971       Fullmetal Alchemist: Brotherhood   9.19    1438767.0  16806.0   
15926  Shingeki no Kyojin: The Final Season   9.17     288274.0  11061.0   
5683                            Steins;Gate   9.11     989905.0   5255.0   
14963    Shingeki no Kyojin Season 3 Part 2   9.10     728435.0   4169.0   
9913                               Gintama°   9.10     161812.0   4508.0   
...                                     ...    ...          ...      ...   
4204                           Tsui no Sora   2.26       2489.0   1773.0   
387                         Hametsu no Mars   2.23      37641.0  22048.0   
15172                                Ex-Arm   2.18      20376.0  15346.0   
6967                     Utsu Musume Sayuri   2.01      10242.0   7196.0   
2934           Tenkuu Danzai Skelter+Heaven   1.85      16946.0  11816.0   

       Score-2  Score-3  Score-4  Score-5  Score-6  Score-7   Score-8  \
3971    2162.0   1536.0   3222.0   9308.0  20210.0  70045.0  199160.0   
15926    514.0    382.0    588.0   1336.0   2674.0   8793.0   26016.0   
5683    1678.0   2292.0   5061.0  11126.0  21375.0  57740.0  140914.0   
14963    385.0    550.0   1108.0   2974.0   8365.0  33662.0  110481.0   
9913    1100.0    648.0    952.0   2311.0   3898.0  10215.0   21360.0   
...        ...      ...      ...      ...      ...      ...       ...   
4204     296.0     90.0     54.0     40.0     24.0     21.0       7.0   
387     7692.0   2648.0   1199.0    648.0    372.0    275.0     183.0   
15172   2014.0    773.0    406.0    235.0    187.0    153.0      51.0   
6967    1246.0    426.0    283.0    288.0    170.0     94.0      54.0   
2934    2770.0    846.0    332.0    173.0    100.0     63.0      33.0   

        Score-9  Score-10  
3971   401507.0  714811.0  
15926   63756.0  173154.0  
5683   275960.0  468504.0  
14963  239451.0  327290.0  
9913    37902.0   78918.0  
...         ...       ...  
4204        5.0     179.0  
387       138.0    2438.0  
15172      56.0    1155.0  
6967       45.0     440.0  
2934       53.0     760.0  

[12116 rows x 13 columns]

In [8]:
#Describes the data contained in the total votes column
animeso1['Total Votes'].describe()

count    1.211600e+04
mean     2.572042e+04
std      8.879405e+04
min      1.010000e+02
25%      5.417500e+02
50%      2.113500e+03
75%      1.218575e+04
max      1.826691e+06
Name: Total Votes, dtype: float64

For the sake of properly defining my standards, a good score is anything above 8.0
And unpopular or unrecognized will mean being below the 25th percentile on 'Total Votes' that is = 542 votes
This will only show results if an anime like that even exists, if not, I will rise the condition to the 50th percentile = 2113

In [23]:
#Filtering out all animes below 8.0 score on myanimelist
animeso1 = animeso1[animeso1['Score'] >= 8.0]
animeso1 = animeso1[animeso1['Total Votes'] <= 542]
animeso1.iloc[:,[1,2,-1,-2,-3,-4,-5,-6,-7,-8,-9,-10,-11]]

Empty DataFrame
Columns: [Name, Score, Total Votes, Score-1, Score-2, Score-3, Score-4, Score-5, Score-6, Score-7, Score-8, Score-9, Score-10]
Index: []

In [9]:
#No animes showed up, lets increase the condition to 50th percentile
#IMPORTANT : for the next step, the kernel was reset, and the last code cell was not run for the sake of not altering animeso1
animeso1 = animeso1[animeso1['Score'] >= 8.0]
animeso1 = animeso1[animeso1['Total Votes'] <= 2113]
animeso1.iloc[:,[1,2,-1,-2,-3,-4,-5,-6,-7,-8,-9,-10,-11]]

Name  Score  Total Votes  Score-1  \
16030         Luo Xiao Hei Zhan Ji (Movie)   8.35       2026.0     16.0   
17373                       Pui Pui Molcar   8.33       1299.0     30.0   
17008           Douluo Dalu: Xiaowu Juebie   8.24       1249.0     48.0   
17007         Douluo Dalu: Qian Hua Xi Jin   8.15       1647.0     15.0   
16156                      Mo Dao Zu Shi Q   8.14       1921.0     37.0   
17006       Douluo Dalu: Hao Tian Yang Wei   8.06       1703.0     16.0   
3882   Ramayana: The Legend of Prince Rama   8.04       1955.0     26.0   

       Score-2  Score-3  Score-4  Score-5  Score-6  Score-7  Score-8  Score-9  \
16030      3.0      8.0     21.0     30.0     83.0    254.0    463.0    622.0   
17373      3.0      6.0      6.0     33.0     62.0    183.0    203.0    159.0   
17008      8.0     10.0     10.0     15.0     42.0    107.0    278.0    304.0   
17007      2.0      2.0     11.0     29.0     90.0    256.0    501.0    352.0   
16156     12.0     11.0     16.0     60.0    109.0    312.0    432.0    345.0   
17006      3.0      4.0     11.0     39.0     91.0    300.0    537.0    333.0   
3882      12.0     13.0     22.0     68.0    139.0    325.0    455.0    315.0   

       Score-10  
16030     526.0  
17373     614.0  
17008     427.0  
17007     389.0  
16156     587.0  
17006     369.0  
3882      580.0

### Results of objective 01
The results are meaningful, we've found 7 animes that have a really small amount of total votes and at the same time have high averages, there could be some hidden gem in those animes

## Fulfilling objective 02
For this objective, its a little bit trickier, we will also look for a high-ish score (7.5 or above) but more importantly we will look for an anime with a huge amount of bad scores (1-4). That will show us that even with a very high score there is a lot of people that dislike it, making it overrated

In [31]:
#First lets create a column named 'Bad Scores' where we will store the sum of all scores 1 to 4
animeso2['Bad Scores'] = animeso2['Score-1'] + animeso2['Score-2'] + animeso2['Score-3'] + animeso2['Score-4']

In [32]:
#Then lets filter out all animes with less than 7.5 score
animeso2 = animeso2[animeso2['Score'] >= 7.5]

In [33]:
#Displaying the columns we are interested on
animeso2.iloc[:,[1,2,-1,-2,-3,-4,-5]]

Name  Score  Bad Scores  \
3971                    Fullmetal Alchemist: Brotherhood   9.19     23726.0   
15926               Shingeki no Kyojin: The Final Season   9.17     12545.0   
5683                                         Steins;Gate   9.11     14286.0   
14963                 Shingeki no Kyojin Season 3 Part 2   9.10      6212.0   
9913                                            Gintama°   9.10      7208.0   
...                                                  ...    ...         ...   
246                                             Hellsing   7.50     11665.0   
6608    Toaru Majutsu no Index Movie: Endymion no Kiseki   7.50      1614.0   
4081    Tales of Symphonia The Animation: Tethe'alla-hen   7.50       314.0   
13841                              Kakuriyo no Yadomeshi   7.50      1430.0   
6814   Initial D: Project D to the Next Stage - Proje...   7.50        76.0   

       Score-1  Score-2  Score-3  Score-4  
3971   16806.0   2162.0   1536.0   3222.0  
15926  11061.0    514.0    382.0    588.0  
5683    5255.0   1678.0   2292.0   5061.0  
14963   4169.0    385.0    550.0   1108.0  
9913    4508.0   1100.0    648.0    952.0  
...        ...      ...      ...      ...  
246      770.0   1184.0   2617.0   7094.0  
6608      84.0    141.0    382.0   1007.0  
4081      28.0     24.0     69.0    193.0  
13841    109.0    138.0    311.0    872.0  
6814      24.0      6.0     15.0     31.0  

[1594 rows x 7 columns]

In [34]:
#Next we want to see the details of the Bad Scores column
animeso2['Bad Scores'].describe()

count     1594.000000
mean      2462.881430
std       4943.251207
min          7.000000
25%        287.000000
50%        857.500000
75%       2357.000000
max      67092.000000
Name: Bad Scores, dtype: float64

In [35]:
#As we can see, there's an anime with over 67 thousand bad scores, thats the one we want to see, lets find it!
animeso2 = animeso2.sort_values(by = 'Bad Scores', ascending = False)

In [36]:
#Displaying the columns we are interested on
animeso2.iloc[:,[1,2,-1,-2,-3,-4,-5]]

Name  Score  Bad Scores  \
6295                                         Mirai Nikki   7.54     67092.0   
4707                                          Fairy Tail   7.68     56246.0   
8625                                      Akame ga Kill!   7.53     52745.0   
202                                           Elfen Lied   7.56     45751.0   
8646                                         Tokyo Ghoul   7.81     45027.0   
...                                                  ...    ...         ...   
16837                                 Inochi no Tabekata   7.69        21.0   
13050  Donten ni Warau Gaiden: Shukumei, Soutou no Fuuma   7.52        19.0   
16574  Fanren Xiu Xian Chuan Zhi Fanren Feng Qi Tian Nan   7.50        18.0   
17052                                    Wan Sheng Jie 2   7.84        15.0   
17505                                          Heikousen   7.52         7.0   

       Score-1  Score-2  Score-3  Score-4  
6295    7443.0   9683.0  16903.0  33063.0  
4707    7090.0   7680.0  13627.0  27849.0  
8625    4214.0   6361.0  12722.0  29448.0  
202     5519.0   7416.0  11087.0  21729.0  
8646    4794.0   5144.0  10284.0  24805.0  
...        ...      ...      ...      ...  
16837      4.0      2.0      1.0     14.0  
13050      4.0      2.0      4.0      9.0  
16574      5.0      4.0      5.0      4.0  
17052      6.0      3.0      2.0      4.0  
17505      1.0      2.0      1.0      3.0  

[1594 rows x 7 columns]

### Results of objective 02
We found the one we were looking for, numbers don't lie, and apparently Mirai Nikki is extremely hated even tho it has a high score, I will consider it overrated

## Fulfilling objective 03
It might seem repetitive, and only time will tell if it is or not, to figure out the most hated anime, i am thinking of finding out the highest bad scores (1-3) to good scores (8-10) ratio, excluding 4,5,6 and 7 allows us to filter out Mid animes

In [37]:
# First we are going to create 2 new columns, one for the sum of 1-3 scores and one for the sum of 8-10 scores
animeso3['Bad Scores'] = animeso3['Score-1'] + animeso3['Score-2'] + animeso3['Score-2']
animeso3['Good Scores'] = animeso3['Score-10'] + animeso3['Score-9'] + animeso3['Score-8']

In [38]:
#Lets display the changes with some other columns
animeso3.iloc[:,[1,2,-1,-2]]

Name  Score  Good Scores  Bad Scores
3971       Fullmetal Alchemist: Brotherhood   9.19    1315478.0     21130.0
15926  Shingeki no Kyojin: The Final Season   9.17     262926.0     12089.0
5683                            Steins;Gate   9.11     885378.0      8611.0
14963    Shingeki no Kyojin Season 3 Part 2   9.10     677222.0      4939.0
9913                               Gintama°   9.10     138180.0      6708.0
...                                     ...    ...          ...         ...
4204                           Tsui no Sora   2.26        191.0      2365.0
387                         Hametsu no Mars   2.23       2759.0     37432.0
15172                                Ex-Arm   2.18       1262.0     19374.0
6967                     Utsu Musume Sayuri   2.01        539.0      9688.0
2934           Tenkuu Danzai Skelter+Heaven   1.85        846.0     17356.0

[12116 rows x 4 columns]

In [39]:
# Now let create another column where we will divide good scores by bad scores, since by this logic, the lower the number the most hated the anime
animeso3['Hated Value'] = animeso3['Good Scores']/animeso3['Bad Scores']

In [40]:
#Lets display the changes with some other columns
animeso3.iloc[:,[1,2,-1,-2,-3]]

Name  Score  Hated Value  Good Scores  \
3971       Fullmetal Alchemist: Brotherhood   9.19    62.256413    1315478.0   
15926  Shingeki no Kyojin: The Final Season   9.17    21.749193     262926.0   
5683                            Steins;Gate   9.11   102.819417     885378.0   
14963    Shingeki no Kyojin Season 3 Part 2   9.10   137.117230     677222.0   
9913                               Gintama°   9.10    20.599284     138180.0   
...                                     ...    ...          ...          ...   
4204                           Tsui no Sora   2.26     0.080761        191.0   
387                         Hametsu no Mars   2.23     0.073707       2759.0   
15172                                Ex-Arm   2.18     0.065139       1262.0   
6967                     Utsu Musume Sayuri   2.01     0.055636        539.0   
2934           Tenkuu Danzai Skelter+Heaven   1.85     0.048744        846.0   

       Bad Scores  
3971      21130.0  
15926     12089.0  
5683       8611.0  
14963      4939.0  
9913       6708.0  
...           ...  
4204       2365.0  
387       37432.0  
15172     19374.0  
6967       9688.0  
2934      17356.0  

[12116 rows x 5 columns]

In [41]:
#lets see more information on the Bad Scores column
animeso3['Bad Scores'].describe()

count    12116.000000
mean       506.777649
std       2029.979130
min          3.000000
25%         37.000000
50%         92.000000
75%        300.000000
max      74214.000000
Name: Bad Scores, dtype: float64

In [42]:
#Let's sort by the ratio and see what we've got
animeso3 = animeso3.sort_values(by = 'Hated Value')

In [43]:
#Lets display the relevant information
animeso3.iloc[:,[1,2,-1,-2,-3]]

Name  Score  Hated Value  \
5089   Atelier Petros Joukuu Gekijou: Sentaku Shima n...   4.31     0.029915   
8620                                     Madonna (Movie)   3.19     0.030588   
8612                                         Scrap Diary   3.32     0.036893   
10259                                         Count Down   4.17     0.038136   
6655                             Within the Bloody Woods   3.39     0.038462   
...                                                  ...    ...          ...   
14617                Aggressive Retsuko (ONA) 2nd Season   7.79   448.095238   
14857                    Wotaku ni Koi wa Muzukashii OVA   7.86   452.358491   
8063                              Gin no Saji 2nd Season   8.29   455.045455   
163                               Initial D Second Stage   8.12   477.851562   
479                                   Majo no Takkyuubin   8.24   553.102719   

       Good Scores  Bad Scores  
5089           7.0       234.0  
8620          13.0       425.0  
8612          19.0       515.0  
10259          9.0       236.0  
6655          34.0       884.0  
...            ...         ...  
14617      28230.0        63.0  
14857      23975.0        53.0  
8063       60066.0       132.0  
163        61165.0       128.0  
479       183077.0       331.0  

[12116 rows x 5 columns]

### Results of objective 3
We got it! not by score but by a pure ratio, we found that Atelier Petros Joukuu Gekijou is the most hated anime

## Fulfilling objective 04
Objective 04 is the polar opposite of objective 03 however the process is the same, so we can agree that the only relevant change would be to flip the dataframe acquired in the last step of objective 03

In [47]:
animeso4 = animeso3
#Lets rename the Hated Value column since it has lost meaning
column_mapper = {'Hated Value':'Loved Value'}
animeso4 = animeso4.rename(
    mapper = column_mapper,
    axis = 1)

In [49]:
#Lets resort the table
animeso4 = animeso4.sort_values(by = 'Loved Value', ascending = False)

In [50]:
#Lets display the changes
animeso4.iloc[:,[1,2,-1,-2,-3]]

Name  Score  Loved Value  \
479                                   Majo no Takkyuubin   8.24   553.102719   
163                               Initial D Second Stage   8.12   477.851562   
8063                              Gin no Saji 2nd Season   8.29   455.045455   
14857                    Wotaku ni Koi wa Muzukashii OVA   7.86   452.358491   
14617                Aggressive Retsuko (ONA) 2nd Season   7.79   448.095238   
...                                                  ...    ...          ...   
6655                             Within the Bloody Woods   3.39     0.038462   
10259                                         Count Down   4.17     0.038136   
8612                                         Scrap Diary   3.32     0.036893   
8620                                     Madonna (Movie)   3.19     0.030588   
5089   Atelier Petros Joukuu Gekijou: Sentaku Shima n...   4.31     0.029915   

       Good Scores  Bad Scores  
479       183077.0       331.0  
163        61165.0       128.0  
8063       60066.0       132.0  
14857      23975.0        53.0  
14617      28230.0        63.0  
...            ...         ...  
6655          34.0       884.0  
10259          9.0       236.0  
8612          19.0       515.0  
8620          13.0       425.0  
5089           7.0       234.0  

[12116 rows x 5 columns]

## Results objective 04
We've found it as well, this time easily
the most loved anime is Majo no Takkyuubin with 553 times the amount of good scores compared to bad scores!